com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -361,7 +361,7 @@ public static void NetworkHide(List<NetworkObject> networkObjects, ulong clientI
 
         private void OnDestroy()
         {
-            if (NetworkManager.Singleton != null)
+            if (NetworkManager.Singleton != null &&  NetworkSpawnManager.SpawnedObjects.Count > 0 && NetworkSpawnManager.SpawnedObjects.ContainsKey(NetworkObjectId))
```

Comment:
`count > 0` check shouldn't be necessary as it is always true if `ContainsKey` returns true right?

Reply:
Checking an Int value first is faster than checking a hashed set, so it is just an early out optimization.  Not required.
Of course, if it is > then you have the int compare plus the hashed set check... :/

50/50 on this.. :)

Reply:
Went ahead and removed it in 5e04e7e.

Reply:
Oh yeah I see I guess yours is much faster. I hoped dictionary would make an empty check with contains key but apparently it doesn't.